pkgs/applications/emulators/dosbox-x/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, file
+, fluidsynth
+, freetype
+, glib
+, libpcap
+, libpng
+, libslirp
+, libXext
+, libxkbfile
+, libXrandr
+, nasm
+, ncurses
+, pkg-config
+, SDL2
+, which
+, yad
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dosbox-x";
+  version = "2023.09.01";
+
+  src = fetchFromGitHub {
+    owner = "joncampbell123";
+    repo = "dosbox-x";
+    rev = "dosbox-x-v${finalAttrs.version}";
+    hash = "sha256-gEqIPXrq16p6VC9aGW2sRSWE0SG9hiMWeRzSoDvUbB0=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    file
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    fluidsynth
+    freetype
+    glib
+    libpcap
+    libpng
+    libslirp
+    libXext
+    libxkbfile
+    libXrandr
+    nasm
+    ncurses
+    SDL2
+    which
```

Comment:
Should be in `nativeBuildInputs`, but see earlier note about `which` not really mattering here.